configure

```diff
@@ -1966,31 +1966,45 @@ esac
 
 # Toolchain for Android is in NDK and is using different naming convention
 if test "$_host_os" = android; then
-	case $_host_cpu in
-	arm)
-		_android_target="armv7a-linux-androideabi16"
-		;;
-	aarch64)
-		# Platform version 21 is needed as earlier versions of platform do not support this architecture.
-		_android_target="aarch64-linux-android21"
-		;;
-	i686)
-		_android_target="i686-linux-android16"
-		;;
-	x86_64)
-		# Platform version 21 is needed as earlier versions of platform do not support this architecture.
-		_android_target="x86_64-linux-android21"
-		;;
-	esac
+	# Try to use a known to work (on linux) toolchain
+	_android_toolchain="$ANDROID_NDK_ROOT/toolchains/llvm/prebuilt/linux-x86_64"
+	if test -n "$ANDROID_TOOLCHAIN"; then
+		_android_toolchain="$ANDROID_TOOLCHAIN"
+	fi
+
+	# If CXX environment variable is not set, try to set known to work defaults
+	if test -z "$CXX"; then
```

Comment:
i was intentionally trying to avoid this, as i didnt find any real-case scenario. i'm not strictly against, but if you know such case, please tell me :)

Reply:
It is really useful for the buildbot as we use ccache there to speed up rebuilds.
To make it work we define CXX="ccache <real compiler>"

Reply:
that is very good use case which i didnt think of, thanks:)